File: MyHordesOptimizerWebsite/src/app/_abstract_model/dto/town-list-item.dto.ts
import { TownPhase, TownTypeId } from '../types/_types';

export interface TownPublicCitizenDTO {
    id: number;
    name: string | null;
    deathTypeId: number | null;
}

export interface TownListItemDTO {
    id: number;
    mapId: number | null;
    name: string | null;
    width: number | null;
    height: number | null;
    townType: TownTypeId | null;
    season: number | null;
    phase: TownPhase | null;
    language: string | null;
    score: number | null;
    isChaos: boolean;
    isDevasted: boolean;
    isFinished: boolean;
    citizens: TownPublicCitizenDTO[];
}

File: MyHordesOptimizerApi/MyHordesOptimizerApi/Dtos/MyHordesOptimizer/TownListItemDto.cs
using System.Collections.Generic;

namespace MyHordesOptimizerApi.Dtos.MyHordesOptimizer
{
    public class TownListItemDto
    {
        public int Id { get; set; }
        public int? MapId { get; set; }
        public string? Name { get; set; }
        public int? Width { get; set; }
        public int? Height { get; set; }
        public TownType? TownType { get; set; }
        public int? Season { get; set; }
        public TownPhase? Phase { get; set; }
        public string? Language { get; set; }
        public int? Score { get; set; }
        public bool IsChaos { get; set; }
        public bool IsDevasted { get; set; }
        public bool IsFinished { get; set; }
        public List<TownPublicCitizenSimpleDto> Citizens { get; set; } = new();
    }
}